Statistics3-3

VladimirVinageras opened the pull request.

VladimirVinageras committed:
Improving code after review in github

VladimirVinageras committed:
Searching issues with 403 error message

VladimirVinageras committed:
Searching issues with 403 error message

VladimirVinageras committed:
Added StatisticsPresenter
Added Network layout for Users
Refactored the StatisticsViewController
StatisticsScreen is Ready

VladimirVinageras committed:
Added StatisticsPresenter
Added Network layout for Users
Refactored the StatisticsViewController
StatisticsScreen is Ready

VladimirVinageras committed:
Added UserCardPresenter
Refactored the UserCardViewController
UserCardScreen is Ready

VladimirVinageras committed:
Inserting other epics in project

VladimirVinageras committed:
UserCollectionScreen finished
Added real data
Added  presenter and services

VladimirVinageras committed:
UserCollectionScreen finished
Added real data
Added  presenter and services

VladimirVinageras committed:
Updated Statistics.md file

VladimirVinageras committed:
Update .gitignore

VladimirVinageras committed:
ready to pull

VladimirVinageras committed:
finishing

andrew-kuzn-yndx commented on FakeNFT/DataProvider/CatalogDataProvider.swift:
```diff
@@ -0,0 +1,83 @@
+
+// MARK: - Final Class
+
+final class CatalogDataProvider: CatalogDataProviderProtocol {
```
Кажется, этого файла нет в проекте. Давай его удалим, если он не нужен?

andrew-kuzn-yndx commented on FakeNFT/DataProvider/NFTTableViewRequest.swift:
```diff
@@ -0,0 +1,19 @@
+    var endpoint: URL?
+    
+    init() {
+        guard let endpoint = URL(string: "https://d5dn3j2ouj72b0ejucbl.apigw.yandexcloud.net/api/v1/collections") else { return }
```
Давай хранить этот юрл тоже в RequestConstants

andrew-kuzn-yndx commented on FakeNFT/DataProvider/ProfileUpdateRequest.swift:
```diff
@@ -0,0 +1,43 @@
+
+import Foundation
+
+struct ProfileUpdateRequest: NetworkRequest {
```
И этого файла нет в проекте XCode

andrew-kuzn-yndx commented on FakeNFT/Scenes /Statistics/Users Collection/UserCollectionNetworkService.swift:
```diff
@@ -0,0 +1,88 @@
+        ProgressHUD.show()
+        self.userCollection = []
+        
+        let dispatchGroup = DispatchGroup()
```
Молодец, что пробуешь работать с GCD!

andrew-kuzn-yndx commented on FakeNFT/Scenes /Statistics/Users Collection/UserCollectionViewController.swift:
```diff
@@ -77,12 +78,14 @@ final class UserCollectionViewController: UIViewController {
             make.trailing.equalTo(view.snp.trailing).inset(16)
-            make.height.equalTo(collectionHeight)
+            make.bottom.equalTo(view.snp.bottom)
+           // make.height.equalTo(collectionHeight)
```
Давай это удалим)

andrew-kuzn-yndx commented on FakeNFT/Scenes /Statistics/Users Collection/UserCollectionViewController.swift:
```diff
@@ -134,6 +139,25 @@ extension UserCollectionViewController : UICollectionViewDelegateFlowLayout , UI
+    
+    func addToCartButtonTapped(cell: StatisticsUserNFTCollectionViewCell) {
+     //   guard let nftModel = cell.getNftModel() else { return }
+     //   presenter?.toggleCartStatus(model: nftModel)
```
Старайся не оставлять такое, правилом хорошего тона является чистить код от лишнего перед сдачей работы. В основную ветку проекта должен попадать только чистый код без лишних принтов и закомментированных блоков)

VladimirVinageras replied:
Привет, Андей.
Отвечу здесь потому что не нашел другой вариант. На последне ревью было замечание по поводу картинка в placeholder когда у пользователя нет Nft, и в рекомендациях сказал чтобы сделал как в figma,  но в макете нет предложенный вариант для такой случае. Подскажи, пожалуйста, как должна выглядит заглушка.

VladimirVinageras committed:
ghost files

VladimirVinageras committed:
ghost files

VladimirVinageras committed:
fixing branch

VladimirVinageras committed:
fixing branch ...

VladimirVinageras committed:
Branch repaired

VladimirVinageras committed:
fixed issues with ghost files
fixed issues with progress indicator
fixed issues with viewCollectionCell interface

VladimirVinageras committed:
adding view

VladimirVinageras committed:
Optimized code

VladimirVinageras committed:
Issues related to UserCollectionView fixed

VladimirVinageras committed:
Issues related to UserCollectionView fixed

VladimirVinageras committed:
Issues related to UserCollectionView fixed

VladimirVinageras committed:
testing commit
